03_map_interface_method/homework/main.go

```diff
@@ -0,0 +1,348 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"strconv"
+)
+
+//https://gist.github.com/bemasher/1777766
+type Stack struct {
+	top  *Element
+	size int
+}
+
+type Element struct {
+	value interface{}
+	next  *Element
+}
+
+func (s *Stack) Len() int {
+	return s.size
+}
+
+// Push a new element onto the stack
+func (s *Stack) Push(value interface{}) {
+	s.top = &Element{value, s.top}
+	s.size++
+}
+
+// Remove the top element from the stack and return it's value
+// If the stack is empty, return nil
+func (s *Stack) Pop() (value interface{}) {
+	if s.size > 0 {
+		value, s.top = s.top.value, s.top.next
+		s.size--
+		return
+	}
+	return nil
+}
+
+//Взять элемент с вершины стека, не извлекая его
+func (s *Stack) Top() (value interface{}) {
+	if s.size > 0 {
+		value = s.top.value
+		return
+	}
+	return nil
+}
+
+//Описание алгоритма
+//http://www.zrzahid.com/convert-to-reverse-polish-notation-and-evaluate-the-expression-shunting-yard-algorithm/
+type Calc interface {
+	Calculate(expression string) (float64, int)
+}
+
+type _calc_ struct{}
```

Comment:
Type name could begin with `_` but should not in most cases. I would say it is generally allowed in the generated code but not in handwritten.